### ISSUE TITLE
Detonate and other similar enchants will not give blocks with the FactionsUUID plugin

### ISSUE THREAD

**Issue by Traizel** [GC-spigot/AdvancedEnchantments#3313]
### Describe the bug

I have an error regarding the advanced enchants and FactionsUUID plugins, when a player uses a pickaxe with the detonate enchant (or any other similar enchants the affect how you get the blocks), they are not receiving the blocks and this error shows in the console:

`[13:10:26 ERROR]: Could not pass event BlockBreakEvent to AdvancedEnchantments v9.0.7
java.lang.NoSuchMethodError: 'com.massivecraft.factions.struct.Relation com.massivecraft.factions.FPlayer.getRelationToLocation()'
	at net.advancedplugins.ae.impl.utils.hooks.plugins.FactionsUUIDHook.getRelationOfLand(FactionsUUIDHook.java:30) ~[AdvancedEnchantments-9.0.7.jar:?]
	at net.advancedplugins.ae.impl.effects.effects.mechanics.protection.external.FactionsCheck.canBreak(FactionsCheck.java:24) ~[AdvancedEnchantments-9.0.7.jar:?]
	at net.advancedplugins.ae.impl.effects.effects.mechanics.protection.ProtectionHandler.lambda$canBreak$0(ProtectionHandler.java:45) ~[AdvancedEnchantments-9.0.7.jar:?]
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?]
	at java.util.HashMap$ValueSpliterator.tryAdvance(HashMap.java:1800) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:637) ~[?:?]
	at net.advancedplugins.ae.impl.effects.effects.mechanics.protection.ProtectionHandler.canBreak(ProtectionHandler.java:45) ~[AdvancedEnchantments-9.0.7.jar:?]
	at net.advancedplugins.ae.impl.effects.effects.actions.handlers.DropsHandler.handle(DropsHandler.java:131) ~[AdvancedEnchantments-9.0.7.jar:?]
	at net.advancedplugins.ae.impl.effects.effects.actions.ActionExecutionBuilder.buildAndExecute(ActionExecutionBuilder.java:71) ~[AdvancedEnchantments-9.0.7.jar:?]
	at net.advancedplugins.ae.impl.effects.effects.mechanics.triggers.internal.MiningTrigger.onBlockBreak(MiningTrigger.java:80) ~[AdvancedEnchantments-9.0.7.jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor91.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:git-Paper-404]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:672) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.level.ServerPlayerGameMode.destroyBlock(ServerPlayerGameMode.java:369) ~[?:?]
	at net.minecraft.server.level.ServerPlayerGameMode.destroyAndAck(ServerPlayerGameMode.java:328) ~[?:?]
	at net.minecraft.server.level.ServerPlayerGameMode.handleBlockBreakAction(ServerPlayerGameMode.java:292) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handlePlayerAction(ServerGamePacketListenerImpl.java:1871) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.handle(ServerboundPlayerActionPacket.java:42) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.a(ServerboundPlayerActionPacket.java:15) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1341) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1318) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1311) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1289) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1177) ~[paper-1.19.3.jar:git-Paper-404]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-404]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]`

How can I have this resolved?

I have reached out to the factionsUUID dev as well for answers

NOTE: With the server log, ignore the heightmap errors, I just upgraded the server from 1.19.3 

### How to reproduce

- PaperMC last version of 1.194 (#521)
- Java 17
- Latest version of AE installed (9.0.8)
- Latest version of FactionsUUID installed (1.6.9.5-U0.6.29-b322)
- Make sure Factions is configured to work on the default world
- Op yourself
- Give yourself a diamond pickaxe
- Enchant the pick with detonate IX
- Find a patch of stone and mine it
- You will notice the blocks do not drop, and an error is shown in the console
- NOTE: If you mine other blocks like dirt, this error does not happen
- NOTE: Other similar enchants can cause this issue, like telepathy or auto smelt, but they also may work as expected

### Screenshots / Videos

https://www.youtube.com/watch?v=v7UJjkF8twU

This video will be available for a week after this bug report or until you guys get back to me

### "/ae plinfo" link

https://paste.md-5.net/bosovoyace

### Server Log

https://pastes.dev/0uTArSBabs

**Comment by mbax** [comment]
FactionsUUID maintainer here!

`com.massivecraft.factions.struct.Relation` has not existed for a long time, and was removed two major releases ago in September 2019. Set your dependency to version 0.6.28 for success. 😄 

Maven info:

```xml
<repository>
	<id>moss-repo</id>
	<url>https://ci.ender.zone/plugin/repository/everything/</url>
</repository>
```

```xml
<dependency>
	<groupId>com.massivecraft</groupId>
	<artifactId>Factions</artifactId>
	<version>1.6.9.5-U0.6.28</version>
	<scope>provided</scope>
</dependency>
```

**Comment by Traizel** [comment]
I would like to point out there are other enchants that produce a similar issue. The enchant "Obsidian Destroyer" produces a similar error, and it looks to be related to FactionsUUID. 

Here is a link to the form I posted on the Discord - 
https://discord.com/channels/376425582139015168/1102440883464458312/1102440883464458312

**Comment by GC-spigot** [comment]
Should be resolved! Let me know if it is for you, in 9.1.3

**Comment by Traizel** [comment]
Looks like it has been resolved. Thank you!
